common.mk

```diff
@@ -139,15 +152,29 @@ else
 
 endif
 
+# compiler=clang Enables compilation with clang. 
+
+ifeq ($(compiler), clang)
```

Comment:
Nitpick: Imagine if someone wanted to use a specific version of clang, e.g. clang-4.0 or clang-3.9.

Reply:
IMO the build is so simple here, that it's not worth bothering parametrizing every little knob. If it's needed, it's easy enough to add another config variable to specify specific clang path. At the moment the makefiles don't do that for nvcc or CUDA version, so doing that for clang is an overkill, IMO.

Reply:
As for performance, clang is somewhat slower than nvcc on most of the tests, ties or wins some.
Here's the run (on P100) of the tests compiled by nvcc and clang: https://pastebin.com/bfWQ9fbv

It's not exactly apples-to-apples comparison as it appears that the data is partitioned somewhat differently due to run-time choices that depend on the code of the kernels generated by compiler (i.e. number of registers used). The heuristics may need to be changed/tuned for clang.